### PR TITLE
Add FXIOS-7421 [v119] #16458 Show error if fakespot does not analyze reviews for product

### PR DESCRIPTION
**Pull request by razvanlitianu**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-7421)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/16458)

## :bulb: Description

| Default | A11y |
|--------|--------|
| image | image | 

## :pencil: Checklist
You have to check all boxes before merging
- [ ] Filled in the above information (tickets numbers and description of your work)
- [ ] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed I updated documentation / comments for complex code and public methods

### CODE REVIEW

**Comment by razvanlitianu** [comment]
Can be tested with the [link](https://www.amazon.com/Amazon-Gift-Card-Birthday-Cupcake/dp/B09NKLXZ5V/ref=mp_s_a_1_2?crid=2VU5QSKQT4A91&keywords=gift+cards+amazon+ecard&qid=1695117112&sprefix=gift%2Caps%2C231&sr=8-2)

File: Client/Frontend/Fakespot/FakespotViewModel.swift [hunk 1]
```diff
@@ -19,19 +19,26 @@ class FakespotViewModel {
...
+                let baseElements = [
+                    ViewElement.reliabilityCard,
                     .adjustRatingCard,
                     .highlightsCard,
                     .qualityDeterminationCard,
-                    .settingsCard]
+                    .settingsCard
+                ]
+                if product?.productId == nil {
+                    elements = [.productCannotBeAnalyzed] + baseElements
```
Comment (thatswinnie) [inline Client/Frontend/Fakespot/FakespotViewModel.swift:31]: Base elements are only two cards: `[.qualityDeterminationCard, .settingsCard]` (see Figma). It's basically the same as the error state. Any chance we can get the state `error` in this case instead?

Comment (razvanlitianu) [reply Client/Frontend/Fakespot/FakespotViewModel.swift:31]: Sadly error means it was a problem with the API, in this case the call is with succes, but the indicator that product cannot be analysed is the nil id. A bit strange, but we can discuss with desktop.

File: Client/Frontend/Fakespot/FakespotViewController.swift [hunk 2]
```diff
@@ -223,6 +223,11 @@ class FakespotViewController: UIViewController, Themeable, UIAdaptivePresentatio
             let view: FakespotMessageCardView = .build()
             view.configure(viewModel.noConnectionViewModel)
             return view
+
+        case .productCannotBeAnalyzed:
```
[the remark this answers is not in the document]
Comment (razvanlitianu) [reply Client/Frontend/Fakespot/FakespotViewController.swift:227]: Interesting, maybe we can have separate `warning` and `info` cases, I'll do a refactor and update the PR.

[the remark this answers is not in the document]
Comment (razvanlitianu) [reply Client/Frontend/Fakespot/FakespotViewController.swift]: Ended up with a `messageCard` case that will contain all messages.

File: Client/Frontend/Fakespot/FakespotViewModel.swift (hunk 1 above)
Comment (thatswinnie) [reply Client/Frontend/Fakespot/FakespotViewModel.swift]: Ok. However base elements should still be only the two cards mentioned above when any message card is displayed.

Comment (razvanlitianu) [reply Client/Frontend/Fakespot/FakespotViewModel.swift]: Oh, I see what this is about now. Updated.

**Review by thatswinnie — APPROVED** [review]

**Pull request merged**